operator/podinfo/podinfo_controller.go

```diff
@@ -56,7 +56,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 			return ctrl.Result{}, err
 		}
 		// Pod info does not exist. Create it.
-		return ctrl.Result{}, r.Create(ctx, generatePodInfo(pod))
+		err = r.Create(ctx, generatePodInfo(pod))
+		if errors.IsAlreadyExists(err) {
+			// Sometimes Create fails with "AlreadyExists" error even though the
+			// previous call to Get returned "NotFound" because of a timing issue.
+			// Requeue without returning the error when this happens, otherwise
+			// the controller logs an error.
+			return ctrl.Result{Requeue: true}, nil
```

Comment:
Do we need to requeue?

Reply:
yeah so the idea is to requeue so that the reconciler exercises the update path the next time around, just to make sure the podinfo has all the up-to-date information.

Reply:
sounds good :heavy_check_mark: